qiskit_experiments/library/characterization/t2hahn.py

```diff
@@ -0,0 +1,157 @@
+            these properties.
+
+            This experiment consists of a series of circuits of the form
+
+            .. parsed-literal::
+
+                 ┌─────────┐┌──────────┐┌───────┐┌──────────┐┌──────────┐┌─┐
+            q_0: ┤ RY(π/2) ├┤ DELAY(t) ├┤ RX(π) ├┤ DELAY(t) ├┤ RY(-π/2) ├┤M├
+                 └─────────┘└──────────┘└───────┘└──────────┘└──────────┘└╥┘
+            c: 1/═════════════════════════════════════════════════════════╩═
+                                                                         0
+            for each *t* from the specified delay times
+            and the delays are specified by the user.
+            The circuits are run on the device or on a simulator backend.
+
+        # section: tutorial
+            :doc:`/tutorials/t2ramsey_characterization`
+
+        """
+    __analysis_class__ = T2HahnAnalysis
+
+    @classmethod
+    def _default_experiment_options(cls) -> Options:
+        """Default experiment options.
+
+        Experiment Options:
+            delays (Iterable[float]): Delay times of the experiments.
+            unit (str): Unit of the delay times. Supported units are
+                's', 'ms', 'us', 'ns', 'ps', 'dt'.
+        """
+        options = super()._default_experiment_options()
+
+        options.delays = None
+        options.unit = "s"
+
+        return options
+
+    def __init__(
+        self,
+        qubit: Union[int, Iterable[int]],
+        delays: Union[List[float], np.array],
+        unit: str = "s",
+    ):
+        """
+        **T2 - Hahn Echo class**
+        Initialize the T2 - Hahn Echo class
+        Args:
+            qubit: the qubit under test.
+            delays (List[float)): delay times of the experiments.
+            unit: Optional, time unit of `delays`.
+                Supported units: 's', 'ms', 'us', 'ns', 'ps', 'dt'. The unit is
+                used for both T2Ramsey and for the frequency.
+
+         Raises:
+             QiskitError : Error for invalid input.
+        """
+        # Initialize base experiment
+        super().__init__([qubit])
+        # Set configurable options
+        self.set_experiment_options(delays=delays, unit=unit)
+        self._verify_parameters()
+
+    def _verify_parameters(self):
+        """
+        Verify input correctness, raise QiskitError if needed.
+        Args:
+            qubit: the qubit under test.
+
+        Raises:
+            QiskitError : Error for invalid input.
+        """
+        if any(delay <= 0 for delay in self.experiment_options.delays):
+            raise QiskitError(
+                f"The lengths list {self.experiment_options.delays} should only contain "
+                "positive elements."
+            )
+        if len(set(self.experiment_options.delays)) != len(self.experiment_options.delays):
+            raise QiskitError(
+                f"The lengths list {self.experiment_options.delays} should not contain "
+                "duplicate elements."
```

Comment:
I'm not sure that we don't allow duplicates

Reply:
Is there a reason not to since it is almost equivalent to more shots for a certain delay.
what do you think is better, do we prefer for the user to allow duplicates and that we will be more flexible or it will give the user freedom that we don't want to give?

Reply:
Let's provide the flexibility, because @chriseclectic once had thoughts in favor of separating the shots to separate delays. There was some talking around it, scattered in different platforms, for now I only found this: https://github.com/Qiskit/qiskit-experiments/pull/19#discussion_r613302028.